### PR TITLE
Admin users page

### LINKED ISSUE #120: Implement removing users as admins + UI
- Create API endpoint to remove a user as an admin (change isAdmin for that user to false)
- Implement functionality on the UI to promote and depromote users according to updated Figma https://www.figma.com/design/ThYfb0htuXZB1gt29QDOa8/High-Fidelity-Wireframes?node-id=418-706 
- Should probably create an endpoint in /routes/admin to get admin users to display in the list (GET /admin/users)

### PR DESCRIPTION
**Pull request by amberchow8**
- Closes #120

### CODE REVIEW

File: packages/api/src/routes/admin.route.ts [hunk 1]
```diff
@@ -7,9 +7,9 @@ import { route } from "../util/handlers";
 const router = Router();
 
 router.post(
-  "/promote",
+  "/users/:id",
```
Comment (elvincheng3) [inline packages/api/src/routes/admin.route.ts:10]: what is the full route? POST /admin/users?

File: packages/ui/src/pages/AdminUsersPage/AdminUsersPage.vue [hunk 2]
```diff
@@ -1,30 +1,238 @@
...
 
-const name = ref("");
-const promoteUser = useMutation(adminAPI.promoteUserToAdmin, {});
+const usersToAdd = ref<string[]>([]);
+const searchQuery = ref("");
+const { data: adminUsers, isLoading } = useQuery(
+  ["admins"],
+  adminAPI.getAdminUsers
+);
+const { data: allUsers } = useQuery(["users"], adminAPI.getAllUsers);
```
Comment (elvincheng3) [inline packages/ui/src/pages/AdminUsersPage/AdminUsersPage.vue:21]: Can we also show a loading component inside the dropdown?

File: packages/ui/src/pages/AdminUsersPage/AdminUsersPage.vue [hunk 3]
```diff
@@ -1,30 +1,238 @@
...
+const addAdmin = useMutation(
+  (userId: string) => adminAPI.addUserAsAdmin(userId),
+  {
+    onSuccess: () => {
```
Comment (elvincheng3) [inline packages/ui/src/pages/AdminUsersPage/AdminUsersPage.vue:25]: Is there any error handling? Do you guys use toasts anywhere?

File: packages/api/src/routes/admin.route.ts (hunk 1 above)
Comment (amberchow8) [reply packages/api/src/routes/admin.route.ts:10]: yes it's POST /admin/users

File: packages/ui/src/pages/AdminUsersPage/AdminUsersPage.vue (hunk 3 above)
Comment (amberchow8) [reply packages/ui/src/pages/AdminUsersPage/AdminUsersPage.vue:25]: We're using notifications from element plus and I will add error notifications to adding/removing admins

**Review by elvincheng3 — APPROVED** [review]
LGTM 🚀

**Pull request merged**